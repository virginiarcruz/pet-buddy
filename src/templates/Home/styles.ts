import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { Grid } from 'components/Grid'

export const Container = styled.main`
  ${({ theme: { color } }) => css`
    background-color: ${color.mainBg};
    width: 100%;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 72px 16px 42px 16px;
    ${media.greaterThan('medium')`
      padding: 96px 156px 90px 156px;
    `}
  `}
`
export const Content = styled.section`
  ${({ theme: { color, border } }) => css`
    background-color: ${color.white};
    width: 100%;
    min-height: 200px;
    border-radius: ${border.radius.medium};
    margin-top: 56px;
    padding: 32px 24px;
    ${media.greaterThan('medium')`
      padding: 40px 40px 32px 40px;
      margin-top: 96px;
    `}
  `}
`

export const ContentGrid = styled(Grid)`
  padding: 0 16px;
  ${media.greaterThan('medium')`
    margin-top: 56px;
  `}
`

export const Column = styled.div``
export const TitleColumn = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.size.medium};
    font-weight: ${theme.font.medium};
    color: ${theme.color.darkGray};
    line-height: 19px;
    padding-bottom: 24px;
    text-align: left;
  `}
`
