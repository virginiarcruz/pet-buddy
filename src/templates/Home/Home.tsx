import Logo from 'components/Logo'
import GridRow from 'components/GridRow'

import { Container, Content, ContentGrid, Column, TitleColumn } from './styles'
import PageHeader from 'components/PageHeader'

const Home = () => (
  <Container data-testid="home-container">
    <Logo />
    <Content>
      <PageHeader />
      <ContentGrid>
        <Column />
        <TitleColumn> Name </TitleColumn>
        <TitleColumn>Type</TitleColumn>
        <TitleColumn> Breed </TitleColumn>
        <TitleColumn> Gender </TitleColumn>
        <TitleColumn> Color </TitleColumn>
        <Column />
      </ContentGrid>
      <GridRow />
    </Content>
  </Container>
)

export default Home
