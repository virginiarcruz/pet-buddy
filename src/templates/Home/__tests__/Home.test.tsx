import React from 'react'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Home from '../Home'

describe('<Home />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Home />)

    expect(
      screen.getByRole('img', { name: /Shelter Buddy logo/i })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    renderWithTheme(<Home />)
    expect(screen.getByTestId('home-container')).toHaveStyle({
      'background-color': '#F6F7FA'
    })
  })
})
