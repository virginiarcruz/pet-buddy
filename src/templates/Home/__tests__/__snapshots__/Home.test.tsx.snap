// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the heading 1`] = `
.c1 {
  width: 206px;
  height: 32px;
}

.c8 {
  display: grid;
  grid-template-columns: repeat(7,1fr);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 32px;
}

.c0 {
  background-color: #F6F7FA;
  width: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 72px 16px 42px 16px;
}

.c2 {
  background-color: #FFFFFF;
  width: 100%;
  min-height: 200px;
  border-radius: 12px;
  margin-top: 56px;
  padding: 32px 24px;
}

.c10 {
  font-size: 16px;
  font-weight: 500;
  color: #707183;
  line-height: 19px;
  padding-bottom: 24px;
  text-align: left;
}

.c6 {
  border-radius: 100px;
  background-color: #F47500;
  color: #FFFFFF;
  font-weight: 600;
  font-size: 14px;
  padding: 4px 8px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 25px;
  line-height: 17px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: #000121;
  font-family: 'Inter',-apple-system,sans-serif;
  font-size: 22px;
  font-weight: 600;
  margin-right: 8px;
  line-height: 27px;
}

.c7 {
  padding: 12px;
  background-color: #F4F4F7;
  border-radius: 12px;
  border: 0;
  font-size: 16px;
  font-family: 'Inter',-apple-system,sans-serif;
  color: #A1A2AF;
  line-height: 19px;
}

.c7::-webkit-input-placeholder {
  color: #A1A2AF;
}

.c7::-moz-placeholder {
  color: #A1A2AF;
}

.c7:-ms-input-placeholder {
  color: #A1A2AF;
}

.c7::placeholder {
  color: #A1A2AF;
}

.c3 > input {
  margin-top: 32px;
}

@media (min-width:768px) {
  .c1 {
    width: 283px;
    height: 44px;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 96px 156px 90px 156px;
  }
}

@media (min-width:768px) {
  .c2 {
    padding: 40px 40px 32px 40px;
    margin-top: 96px;
  }
}

@media (min-width:768px) {
  .c9 {
    margin-top: 56px;
  }
}

@media (min-width:768px) {
  .c7 {
    width: 310px;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c3 > input {
    margin-top: 0;
  }
}

<div>
  <main
    class="c0"
    data-testid="home-container"
  >
    <img
      alt="Shelter Buddy logo"
      class="c1"
      src="/img/logo.svg"
    />
    <section
      class="c2"
    >
      <header
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5"
          >
            Your Animals
          </h1>
          <span
            class="c6"
          >
            28
          </span>
        </div>
        <input
          class="c7"
          placeholder="Search an animal by name"
          type="text"
        />
      </header>
      <div
        class="c8 c9"
      >
        <div
          class=""
        />
        <h2
          class="c10"
        >
           Name 
        </h2>
        <h2
          class="c10"
        >
          Type
        </h2>
        <h2
          class="c10"
        >
           Breed 
        </h2>
        <h2
          class="c10"
        >
           Gender 
        </h2>
        <h2
          class="c10"
        >
           Color 
        </h2>
        <div
          class=""
        />
      </div>
      <div
        class="c11"
      />
    </section>
  </main>
</div>
`;
